service/controller/v3/resource/vnetpeeringcleaner/create.go

```diff
@@ -0,0 +1,43 @@
+package vnetpeeringcleaner
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/azure-operator/service/controller/v3/key"
+)
+
+// EnsureCreated ensure that vnetpeering resource are deleted,
+// since they are no longer in use in this version.
+func (r Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring delete host vnetpeering")
+
+	azureConfig, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	vnetPeeringClient, err := r.getVnetPeeringClient()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	resourceGroupName := r.azure.HostCluster.ResourceGroup
+	vnetName := r.azure.HostCluster.ResourceGroup
+	peeringName := key.ResourceGroupName(azureConfig)
+	respFuture, err := vnetPeeringClient.Delete(ctx, resourceGroupName, vnetName, peeringName)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// DeleteResponder ensure that response body is closed.
+	_, err = vnetPeeringClient.DeleteResponder(respFuture.Response())
```

Comment:
Can you explain why we need this? I do not understand it. IMO it is not necessary.

Reply:
It is to ensure response body is closed, in order to avoid memory leaks.

Reply:
According to azure godoc: "DeleteResponder handles the response to the Delete request`"

We can then handle result with helpers like `client.ResponseWasNotFound`
https://github.com/giantswarm/azure-operator/blob/0e88f6eefa1ddcdd9d080fa32858ce641909b1d0/client/azure_client_set.go#L123-L129

Reply:
If this is true we have these memory leaks in several places. I am not into the details of the SDK implementation but I would guess there is not leaking anything that you do not make use of.